Sources/RemoteMessaging/Matchers/UserAttributeMatcher.swift

```diff
@@ -136,6 +154,22 @@ public struct UserAttributeMatcher: AttributeMatcher {
             } else {
                 return RangeIntMatchingAttribute(min: matchingAttribute.min, max: matchingAttribute.max).matches(value: privacyProDaysUntilExpiry)
             }
+        case let matchingAttribute as PrivacyProPurchasePlatformMatchingAttribute:
+            return StringArrayMatchingAttribute(matchingAttribute.value).matches(value: privacyProPurchasePlatform ?? "")
+        case let matchingAttribute as PrivacyProSubscriptionStatusMatchingAttribute:
+            guard let value = matchingAttribute.value else {
+                return .fail
+            }
+
+            guard let status = PrivacyProSubscriptionStatus(rawValue: value) else {
+                return .fail
+            }
+
+            switch status {
+            case .active: return isPrivacyProSubscriptionActive ? .match : .fail
+            case .expiring: return isPrivacyProSubscriptionExpiring ? .match : .fail
+            case .expired: return isPrivacyProSubscriptionExpired ? .match : .fail
```

Comment:
Here we don't match against Privacy Pro account states 1:1, but instead have three supported options that match various account states. For example, `active` matches users who have entitlements of any form, even if their account is going to expire later - we still want to match them because they can use the service.